src/control/lib/control/auto.go

```diff
@@ -554,32 +554,50 @@ func defaultEngineCfg(idx int) *engine.Config {
 }
 
 // genConfig generates server config file from details of available network,
-// storage and CPU hardware.
+// storage and CPU hardware after performing some basic sanity checks.
 func genConfig(log logging.Logger, accessPoints []string, nd *networkDetails, sd *storageDetails, ccs numaCoreCountsMap) (*config.Server, error) {
-	// basic sanity checks
 	if nd.engineCount == 0 {
 		return nil, errors.Errorf(errInvalNrEngines, 1, 0)
 	}
+
 	if len(nd.numaIfaces) < nd.engineCount {
 		return nil, errors.Errorf(errInsufNrIfaces, "", nd.engineCount,
 			len(nd.numaIfaces), nd.numaIfaces)
 	}
+
 	if len(sd.numaPMems) < nd.engineCount {
 		return nil, errors.Errorf(errInsufNrPMemGroups, sd.numaPMems, nd.engineCount,
 			len(sd.numaPMems))
 	}
-	if len(sd.numaSSDs) > 0 && (len(sd.numaSSDs) < nd.engineCount) {
-		return nil, errors.New("invalid number of ssd groups") // shouldn't happen
+
+	// enforce consistent ssd count across engine configs
+	minSsds := math.MaxUint32
+	numaWithMinSsds := 0
+	if len(sd.numaSSDs) > 0 {
+		if len(sd.numaSSDs) < nd.engineCount {
+			return nil, errors.New("invalid number of ssd groups") // should never happen
+		}
+
+		for numa, ssds := range sd.numaSSDs {
+			if len(ssds) < minSsds {
+				minSsds = len(ssds)
+				numaWithMinSsds = numa
+			}
+		}
 	}
```

Comment:
If I understand the logic here correctly, the numa node with the least number of SSDs becomes the lowest common denominator, right? So if you have a system with 2 SSDs on numa 0 and none on numa 1, doesn't that mean that the config will be generated with 0 SSDs? I don't think that's what we want... Instead, we probably want to generate a config that only uses a single NUMA node, in that case.

Reply:
The approach currently is that the config generator is trying to create config for a derived number of engines, currently the behaviour in the scenario you specify is that you would get an early validation failure because the number of SSDs is below the acceptable number per-NUMA. Is that acceptable? The remedy is then to run with a lower number of target engines e.g. 1 which should then produce a config.

Reply:
> The approach currently is that the config generator is trying to create config for a derived number of engines, currently the behaviour in the scenario you specify is that you would get an early validation failure because the number of SSDs is below the acceptable number per-NUMA. Is that acceptable? The remedy is then to run with a lower number of target engines e.g. 1 which should then produce a config.

OK, thanks for explaining. I am fine with that approach as long as the failure reason and remedy are very clear from the output displayed to the user.

Reply:
have added test coverage for this case (and improved some of the tests)